lms/djangoapps/grades/tasks.py

```diff
@@ -36,63 +56,57 @@ def recalculate_subsection_grade(
         only_if_higher (boolean): indicating whether grades should
             be updated only if the new raw_earned is higher than the
             previous value.
-        weighted_earned (float): the weighted points the learner earned on the
-            problem that triggered the update.
-        weighted_possible (float): the max weighted points the leaner could have
-            earned on the problem.
+        task_queued_time (serialized datetime): indicates when the task
+            was queued so that we can verify the underlying data update.
         score_deleted (boolean): indicating whether the grade change is
             a result of the problem's score being deleted.
     """
-    course_key = CourseLocator.from_string(course_id)
+    course_key = CourseLocator.from_string(kwargs['course_id'])
     if not PersistentGradesEnabledFlag.feature_enabled(course_key):
         return
 
     score_deleted = kwargs['score_deleted']
-    scored_block_usage_key = UsageKey.from_string(usage_id).replace(course_key=course_key)
+    scored_block_usage_key = UsageKey.from_string(kwargs['usage_id']).replace(course_key=course_key)
+    task_queued_time = datetime.strptime(kwargs['task_queued_time'], "%Y-%m-%d %H:%M:%S.%f").replace(tzinfo=pytz.UTC)
```

Comment:
Please create a common helper for converting from string to datetime object in date_utils.py.